x-pack/plugins/index_management/public/application/services/routing.ts

```diff
@@ -39,3 +33,8 @@ export const decodePath = (pathname: string): string => {
   }
   return decodeURIComponent(decodedPath);
 };
+
+// Need to add some additonal encoding/decoding logic to work with React Router
+// For background, see: https://github.com/ReactTraining/history/issues/505
```

Comment:
Thanks for adding this!

Reply:
To be fair I think the comment was already there! I just moved it around. :)